src/praisonai-agents/praisonaiagents/llm/model_capabilities.py

```
@@ -276,6 +276,44 @@ def supports_web_fetch(model_name: str) -> bool:
+        bool: True if the model is a reasoning model, False otherwise
+    """
+    if not model_name:
+        return False
+
+    try:
+        litellm = _get_litellm()
+        if litellm is not None and hasattr(litellm, 'supports_reasoning'):
+            if litellm.supports_reasoning(model=model_name):
+                return True
+    except Exception:
+        pass
```

Ruff (0.16.1)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)